testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -24,612 +23,419 @@ import (
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
-			}
-			if lResp.Data == nil {
-				return errEmptyLighthouseData
+				return errEmptyData
 			}
 			return nil
-		},
-	},
-	"/beacon/pool/bls_to_execution_changes": {
-		basepath: v1PathTemplate,
-		prysmResps: map[string]interface{}{
-			"json": &beacon.BLSToExecutionChangesPoolResponse{},
-		},
-		lighthouseResps: map[string]interface{}{
-			"json": &beacon.BLSToExecutionChangesPoolResponse{},
-		},
-		customEvaluation: func(p interface{}, l interface{}) error {
+		})),
+	"/beacon/pool/bls_to_execution_changes": newMetadata[beacon.BLSToExecutionChangesPoolResponse](v1PathTemplate,
+		withCustomEval(func(p interface{}, _ interface{}) error {
 			pResp, ok := p.(*beacon.BLSToExecutionChangesPoolResponse)
 			if !ok {
-				return errJsonCast
-			}
-			lResp, ok := l.(*beacon.BLSToExecutionChangesPoolResponse)
-			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &beacon.BLSToExecutionChangesPoolResponse{}, p)
 			}
 			if pResp.Data == nil {
-				return errEmptyPrysmData
-			}
-			if lResp.Data == nil {
-				return errEmptyLighthouseData
+				return errEmptyData
 			}
 			return nil
-		},
-	},
-	"/config/fork_schedule": {
-		basepath: v1PathTemplate,
-		prysmResps: map[string]interface{}{
-			"json": &config.GetForkScheduleResponse{},
-		},
-		lighthouseResps: map[string]interface{}{
-			"json": &config.GetForkScheduleResponse{},
-		},
-		customEvaluation: func(p interface{}, l interface{}) error {
-			// remove all forks with far-future epoch
-			pSchedule, ok := p.(*config.GetForkScheduleResponse)
+		})),
+	"/config/fork_schedule": newMetadata[config.GetForkScheduleResponse](v1PathTemplate,
+		withCustomEval(func(p interface{}, l interface{}) error {
+			pResp, ok := p.(*config.GetForkScheduleResponse)
 			if !ok {
-				return errJsonCast
-			}
-			for i := len(pSchedule.Data) - 1; i >= 0; i-- {
-				if pSchedule.Data[i].Epoch == fmt.Sprintf("%d", params.BeaconConfig().FarFutureEpoch) {
-					pSchedule.Data = append(pSchedule.Data[:i], pSchedule.Data[i+1:]...)
-				}
+				return fmt.Errorf(msgWrongJson, &config.GetForkScheduleResponse{}, p)
 			}
-			lSchedule, ok := l.(*config.GetForkScheduleResponse)
+			lResp, ok := l.(*config.GetForkScheduleResponse)
 			if !ok {
-				return errJsonCast
+				return fmt.Errorf(msgWrongJson, &config.GetForkScheduleResponse{}, l)
+			}
+			// remove all forks with far-future epoch
+			for i := len(pResp.Data) - 1; i >= 0; i-- {
+				if pResp.Data[i].Epoch == fmt.Sprintf("%d", params.BeaconConfig().FarFutureEpoch) {
+					pResp.Data = append(pResp.Data[:i], pResp.Data[i+1:]...)
+				}
 			}
-			for i := len(lSchedule.Data) - 1; i >= 0; i-- {
-				if lSchedule.Data[i].Epoch == fmt.Sprintf("%d", params.BeaconConfig().FarFutureEpoch) {
-					lSchedule.Data = append(lSchedule.Data[:i], lSchedule.Data[i+1:]...)
+			for i := len(lResp.Data) - 1; i >= 0; i-- {
```

Comment:
In almost every other example you have been checking whether `pResp.Data == nil`
Is it not better to check here as well before we start using `pResp.Data` and `lResp.Data`?